eventuate-common-spring-reactive-jdbc/src/main/java/io/eventuate/common/spring/jdbc/reactive/EventuateSpringReactiveJdbcStatementExecutor.java

```diff
@@ -1,14 +1,16 @@
 package io.eventuate.common.spring.jdbc.reactive;
 
 import io.eventuate.common.jdbc.EventuateDuplicateKeyException;
+import io.eventuate.common.reactive.jdbc.EventuateReactiveJdbcStatementExecutor;
+import io.eventuate.common.reactive.jdbc.EventuateReactiveRowMapper;
 import org.springframework.r2dbc.core.DatabaseClient;
 import reactor.core.publisher.Flux;
 import reactor.core.publisher.Mono;
 
 import java.math.BigInteger;
 import java.util.Map;
 
-public class EventuateSpringReactiveJdbcStatementExecutor {
+public class EventuateSpringReactiveJdbcStatementExecutor implements EventuateReactiveJdbcStatementExecutor {
```

Comment:
in insertAndReturnId, we could use s.returnGeneratedValues to get the last insert id.


  public Mono<Long> insertAndReturnId(String sql,
                                      String idColumn,
                                      Object... params) {
    sql = reformatParameters(sql, params);
    DatabaseClient.GenericExecuteSpec genericExecuteSpec = bindParameters(databaseClient.sql(sql)
        .filter(s -> s.returnGeneratedValues(idColumn)), params);

    return genericExecuteSpec
        .map((row, rowMetadata) -> row.get(idColumn, Long.class))
        .first()
        .onErrorResume(t -> Optional.of(t)
            .filter(e -> e instanceof DataIntegrityViolationException)
            .<Mono<Long>>map(e -> Mono.error(new EventuateDuplicateKeyException(e)))
            .orElse(Mono.error(t)));
  }

Reply:
@dartartem Please review this suggestion

Reply:
will do